### PR TITLE
164829214 authenticated users can see author profiles

### PR DESCRIPTION
**Pull request by ValerieRono**
#### Description
When a user is authenticated and verified, he/she should be able to see a list of authors plus their profiles

#### Type of change
- [ ] Bug fix (nonbreaking change which fixes an issue)
- [x] New feature (nonbreaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

#### How Has This Been Tested?
Please describe the tests that you ran to verify your changes
- [ ] End to end
- [x] Integration

### How can this be manually tested?
1.  Clone the repository 
```https://github.com/andela/ah-centauri-backend.git```

2. cd into project directory and checkout into this branch
```bash 
cd ah-centauri-backend
git checkout feature/164829214-list-users-functionality
```

3. Rename the .env.example file into .env and update the parameters with your own keys.

4. Start development server and send the following through postman

The feature has one endpoint, i.e.
 - `GET` all authors plus their profiles - `/articles/authors/profiles/`

#### Checklist:
- [x] My code follows the style guide for this project
- [ ] I have linted my code prior to submission
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

#### Pivotal Tracker
-ensure authenticated and verified users can see profiles of authors
-write tests for said functionality

[Delivers #164829214]

### CODE REVIEW

File: authors/apps/articles/tests/test_get_profiles.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APIClient
+
+from authors.apps.articles.models import Articles
+from authors.apps.authentication.models import User
+
+PROFILES_URL = reverse('articles:author-profile-list')
+
+class ArticleTest(TestCase):
```
Comment (Sm0k3s) [inline authors/apps/articles/tests/test_get_profiles.py]: Change this class name to something related to profiles like --> ProfileTest

**Review by Sm0k3s — APPROVED** [review]

**Review by vino-jasuba — APPROVED** [review]
Great work on this 👍

**Review by bl4ck4ndbr0wn — APPROVED** [review]
Great implementation. 
Good work

File: authors/apps/articles/tests/test_get_profiles.py (hunk 1 above)
Comment (ValerieRono) [reply authors/apps/articles/tests/test_get_profiles.py]: done and done

File: authors/apps/articles/permissions.py [hunk 2]
```diff
@@ -14,3 +14,12 @@ def has_object_permission(self, request, view, obj):
 
         # Write permissions are only allowed to the owner of the snippet.
         return obj.author == request.user
+
+class IsVerified(permissions.BasePermission):
+    """
+    Custom permission to only allow owners of an article to edit its details.
```
Comment (oagutu) [inline authors/apps/articles/permissions.py]: Correct this docstring

File: authors/apps/articles/views.py [hunk 3]
```diff
@@ -625,3 +629,19 @@ def delete(self, request, pk, format=None):
...
+    Allow any authenticated/verified users to hit this endpoint.
+    List all authors plus their profiles.
+    """
+    permission_classes = (IsAuthenticated, IsVerified,)
+    serializer_class = UserSerializer
+    queryset = User.objects.all()
+    
+    def get_queryset(self):
+        queryset = Articles.objects.all()
+        articles =[article.author.id for article in queryset]
```
Comment (oagutu) [inline authors/apps/articles/views.py]: Just a suggestion, you could make use of `values_list` when querying to return a list of `author ids` so you don't have to create a list comprehension

**Review by oagutu — COMMENTED** [review]
@ValerieRono nice work

File: authors/apps/articles/permissions.py (hunk 2 above)
Comment (ValerieRono) [reply authors/apps/articles/permissions.py]: my bad, done and done

File: authors/apps/articles/views.py (hunk 3 above)
Comment (ValerieRono) [reply authors/apps/articles/views.py]: done and done

**Pull request merged**